### ISSUE TITLE
constructor detection for `com.google.gson.Gson` is working but tests unable to detect

### ISSUE THREAD

**Issue by theapache64** [googlesamples/android-custom-lint-rules#96]
**Code: Modified avoid AvoidDateDetector.kt**
```kotlin
override fun getApplicableConstructorTypes(): List<String> = listOf("com.google.gson.Gson")

  override fun visitConstructor(
    context: JavaContext,
    node: UCallExpression,
    constructor: PsiMethod,
  ) {
    context.report(
      ISSUE,
      node,
      context.getLocation(node),
      "Don't use `Gson`;",
    )
  }
```

**Test**
```kotlin
fun testDocumentationExample() {
    lint()
      .files(
        kotlin(
            """
              package test.pkg
              import com.google.gson.Gson
              fun test() {
                val gson = Gson()
              }
              """
          )
          .indented(),
      )
        .allowCompilationErrors()
      .run()
      .expectErrorCount(1)
  }
```

**Result**
- `./gradlew lint` is detecting this but test can't and the above test fails

**Comment by tnorbye** [comment]
It's likely this:
https://googlesamples.github.io/android-custom-lint-rules/api-guide.md.html#frequentlyaskedquestions//mydetectorcallbacksaren'tinvoked

See https://googlesamples.github.io/android-custom-lint-rules/api-guide/unit-testing.md.html#lintcheckunittesting/librarydependenciesandstubs
